cmd/destroy.go

```diff
@@ -151,6 +159,11 @@ func newDestroyCmd() *cobra.Command {
 		&message, "message", "m", "",
 		"Optional message to associate with the destroy operation")
 
+	targets = cmd.PersistentFlags().StringArrayP(
+		"target", "t", []string{},
+		"Specify a single resource URN to destroy. All resources necessary to destroy this target will also be destroyed."+
+			" Multiple resources can be specified using: --target urn1 --target urn2")
```

Comment:
Quite frustratingly, since our URNs often have `$` in them they silently fail if passed unquoted in bash. 

This has tripped up nearly every user who has tried to use similar commands in `Pulumi state`. 

I think we should include quotes around this as will in practice be needed. 

We may even want to add a helpful warning if the name doesn’t match, possibly even doing a similarity comparison to suggest what resource the user actually intended (in all commands which accept a urn string).

Reply:
> We may even want to add a helpful warning if the name doesn’t match

I do issue a message stating if we weren't able to find a resource with that name.

> I think we should include quotes around this as will in practice be needed.

Note: this didn't help for me as even in quotes the shell munged with `$`.  I had to manually escape it even in a string.

Reply:
I believe single quotes are needed:

```
$ echo '$FOO'
$FOO

$ echo "$FOO"


```

Reply:
Gotcha.  yes.  i've updated the PR to at least try to guide people down this path.